django/utils/timezone.py

```diff
@@ -295,13 +295,15 @@ def template_localtime(value, use_tz=None):
 
 # Utilities
 
-def localtime(value, timezone=None):
+def localtime(value=None, timezone=None):
```

Comment:
@aaugustin The current implementation of `localdate` in this PR includes an `if settings.USE_TZ` as an explicit case in order to support both. That makes it inconsistent with `localtime` which supports only `USE_TZ=True`. Are you suggesting that that conditional should be removed from `localdate` in this PR, and then in a separate PR support for `USE_TZ=False` could be added to `localtime`, which would automatically cascade through to `localdate`?

That eventual implementation sounds good to me, though I don't really see a problem doing it all in this PR, rather than removing the support from `localdate` in this PR, which would also entail removing tests that are already here.

I think raising an error if `localtime` receives a naive datetime when `USE_TZ=True` is appropriate.